packages/theme-language-server-common/src/renamed/RenameHandler.ts

```diff
@@ -0,0 +1,34 @@
+import { FileExists, findRoot, path } from '@shopify/theme-check-common';
+import { Connection } from 'vscode-languageserver';
+import { RenameFilesParams } from 'vscode-languageserver-protocol';
+import { DocumentManager } from '../documents';
+import { BaseRenameHandler } from './BaseRenameHandler';
+import { AssetRenameHandler } from './handlers/AssetRenameHandler';
+import { SnippetRenameHandler } from './handlers/SnippetRenameHandler';
+
+/**
+ * The RenameHandler is responsible for handling workspace/didRenameFiles notifications.
+ *
+ * Stuff we'll handle:
+ * - When a snippet is renamed, then we'll change all the render calls
+ * - When an asset is renamed, then we'll change the asset_url calls
+ * - etc.
+ */
+export class RenameHandler {
```

Comment:
Called these handlers because they are not in a request/response cycle. They don't _provide_ anything, they deal with a notification. 

We also already have a RenameProvider and that's for renaming variables/html tags. Different feature similar name.